lib/middleware/parameter.js

```diff
@@ -282,6 +309,33 @@ module.exports = async (ctx, next) => {
                 await Promise.all(tasks);
             }
 
+            // openai
+            if (ctx.query.gpt && config.openai.openai_key) {
+                ctx.state.data.item = await Promise.all(
+                    ctx.state.data.item.map(async (item) => {
+                        if (item.description) {
+                            const summary = await ctx.cache.tryGet(
+                                `openai:${item.link}`,
+                                async () => {
+                                    const text = htmlToText.htmlToText(item.description);
+                                    if (text.length < 300) {
+                                        return '';
+                                    }
+                                    const summary_md = await summarizeArticle(text);
+                                    return marked.parse(summary_md);
+                                },
+                                7 * 24 * 60 * 60 // 7 days
```

Comment:
The cache duration should leave empty which defaults to a configurable `CACHE_CONTENT_EXPIRE`